### ISSUE TITLE
PDF output: usage of PAPER environment variable is broken since Sphinx 1.5

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#5234]
#### Description of problem

Since Sphinx 1.5, more precisely probably commit 72b76ab6d7, and the deprecation of former configuration variable `latex_paper_size` in favour of `latex_elements['papersize']`, the usage of PAPER env variable is broken: LaTeX document is created with `a4` or `letter` option where it should be `a4paper` or `letterpaper`.

This breaks old projects Makefiles following our documentation which says to use `a4` or `letter`.

#### Expected results

Setting `PAPER=a4` or `PAPER=letter` works.

### Reproducible project / your project

Any project.

### Environment info
- OS: Mac
- Python version: 3.6.4
- Sphinx version: 1.5.x and later
- LaTeX

**Comment by jfbu** [comment]
As an example of a broken project consider the CPython documentation: https://github.com/python/cpython/blob/9c18b1ae527346bc178250ad1ca07bffdacde5dd/Doc/Makefile#L1-L1

**Comment by jfbu** [comment]
The CPython [Doc/Makefile](https://github.com/python/cpython/blob/master/Doc/Makefile) contains

```
ALLSPHINXOPTS = -b $(BUILDER) -d build/doctrees -D latex_elements.papersize=$(PAPER) \
                $(SPHINXOPTS) . build/$(BUILDER) $(SOURCES)

build:
	-mkdir -p build
[...cut lines...]
	$(SPHINXBUILD) $(ALLSPHINXOPTS)
	@echo


[...]

latex: BUILDER = latex
latex: build
	@echo "Build finished; the LaTeX files are in build/latex."
	@echo "Run \`make all-pdf' or \`make all-ps' in that directory to" \
	      "run these through (pdf)latex."

[...]

dist:

[...]

	# archive the A4 latex
	rm -rf build/latex
	make latex PAPER=a4
	-sed -i 's/makeindex/makeindex -q/' build/latex/Makefile
	(cd build/latex; make clean && make all-pdf && make FMT=pdf zip bz2)
```

This creates LaTeX files with wrong `a4` option, in place of `a4paper`. It seems this Makefile was updated on model of Sphinx ("old") Makefile at 1.5, at any rate after deprecation of the `latex_paper_size` configuration variable. I will make PR at their project to fix this particular Makefile (updating Sphinx will not help that project if they keep same Makefile.)

**Comment by jfbu** [comment]
See https://github.com/python/cpython/pull/8585 for fix to CPython Doc/Makefile.

**Comment by jfbu** [comment]
Fixed at 077e9af
